Tribler/Core/TorrentChecker/torrent_checker.py

```diff
@@ -111,8 +59,8 @@ def initialize(self):
 
         self._reschedule_torrent_select()
 
-        self._checker_thread = TorrentCheckerThread(self)
-        self._checker_thread.start()
+        self.register_task(u"task_check_sockets", reactor.callLater(self._socket_selection_interval,
```

Comment:
I found that if something goes wrong in `_check_sessions()`, it will not return two proper read_socket and write_socket lists and the thing will break when selecting upon some invalid file descriptor. So I either schedule it when it's ok or I make a LoopingCall in the beginning and when it goes wrong, I cancel the task. Both seem ok so I did the `callLater()` which is faster for me to do.